mllib/src/test/scala/org/apache/spark/ml/classification/LogisticRegressionSuite.scala

```diff
@@ -883,6 +884,27 @@ class LogisticRegressionSuite
     assert(model1a0.intercept ~== model1b.intercept absTol 1E-3)
   }
 
+  test("logistic regression with fitIntercept=true and all labels the same") {
+    val lr = new LogisticRegression()
+      .setFitIntercept(true)
+      .setMaxIter(3)
+    val sameLabels = dataset
+      .withColumn("zeroLabel", lit(0.0))
+      .withColumn("oneLabel", lit(1.0))
+
+    val allZeroModel = lr
+      .setLabelCol("zeroLabel")
+      .fit(sameLabels)
+    assert(allZeroModel.coefficients ~== Vectors.dense(0.0) absTol 1E-3)
+    assert(allZeroModel.intercept === Double.NegativeInfinity)
+
+    val allOneModel = lr
+      .setLabelCol("oneLabel")
+      .fit(sameLabels)
+    assert(allOneModel.coefficients ~== Vectors.dense(0.0) absTol 1E-3)
+    assert(allOneModel.intercept === Double.PositiveInfinity)
+  }
+
```

Comment:
Can you add one test which is all labels the same but `fitIntercept=false` here to avoid the issue in LiR? Thanks.